app/gui/view/graph-editor/src/component/node/input/area.rs

```diff
@@ -102,6 +104,29 @@ impl Debug for Expression {
     }
 }
 
+// === Pretty printing display adapter ===
+
+/// Display adapter used for pretty-printing the `Expression` span tree.
+#[derive(Debug)]
+pub struct ExpressionTreePrettyPrint<'a> {
+    expression: &'a Expression,
+}
+
+impl<'a> Display for ExpressionTreePrettyPrint<'a> {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> fmt::Result {
+        let printed = self.expression.span_tree.debug_print(&self.expression.code);
+        f.write_str(&printed)
+    }
+}
+
+impl Expression {
+    /// Wrap the expression into a pretty-printing adapter that implements `Display` and prints
+    /// detailed span-tree information.
```

Comment:
why expression cant have this pretty printing by default and we need this "adapter"?